messaging/src/main/java/org/axonframework/eventhandling/deadletter/DeadLetteringEventHandlerInvoker.java

```diff
@@ -260,6 +340,42 @@ public Builder allowReset(boolean allowReset) {
             return this;
         }
 
+        /**
+         * Enabled this {@link DeadLetteringEventHandlerInvoker} to cache sequence identifiers. If enabled it will
+         * create a {@link SequenceIdentifierCache} per segment to prevent calling
+         * {@link SequencedDeadLetterQueue#enqueueIfPresent(Object, Supplier)} when we can be sure the sequence
+         * identifier is not present. This can happen in two cases, either we started with an empty
+         * {@link SequencedDeadLetterQueue} and we haven't enqueued this identifier yet. Or it was not empty at the
+         * start, in which case we can skip if we checked if we needed to enqueue the identifier before, and it wasn't
+         * the case. If the identifier might be present we always call the
+         * SequencedDeadLetterQueue#enqueueIfPresent(Object, Supplier)} as the sequence might have been cleaned up in
+         * the meantime.
+         *
+         * @return The current Builder instance for fluent interfacing.
+         */
+        public Builder enableSequenceIdentifierCache() {
+            this.sequenceIdentifierCacheEnabled = true;
+            return this;
+        }
+
+        /**
+         * Sets the size of the cache. When there are already sequences stored in a dead letter queue, we need to check
+         * for each sequence identifier if it's already included. This result is stored in order to we can skip the
+         * check the second time we encounter the same sequence identifier. To limit memory use, a limit is set, by
+         * default at {@code 1024}. If you have a lot of long living aggregates, it might improve performance setting
+         * this higher at the cost of more memory use. If you only have aggregates that are short-lived, setting it
+         * lower frees up memory, while it might not affect performance. This setting is applied per {@link Segment}.
+         * Note that this setting will only be used in combination with {@link #enableSequenceIdentifierCache()}, and
+         * the {@link SequencedDeadLetterQueue} not being empty.
```

Comment:
Relatively minor readability suggestion here:
```suggestion
         * Sets the size of the cache. When there are already sequences stored in a dead-letter queue, we need to check
         * for each sequence identifier if it's already included. This result is stored so we can skip a second "is enqueued" 
         * check when we encounter the same sequence identifier. This method thus limits memory use of the cache. 
         * The size defaults to {@code 1024}. 
         * <p>
         * If you have a lot of long-living aggregates, it might improve performance to increase the
         * cache size at the cost of more memory use. If you only have aggregates that are short-lived, setting it
         * to a lower value frees up memory, while it might not affect performance. 
         * <p>
         * This setting is applied per {@link Segment}.
         * Note that this setting will only be used in combination with {@link #enableSequenceIdentifierCache()}, and
         * when the {@link SequencedDeadLetterQueue} is not empty when constructing the cache.
```